src/herder/QuorumIntersectionCheckerImpl.cpp

```diff
@@ -674,18 +674,19 @@ void
 QuorumIntersectionCheckerImpl::buildSCCs()
 {
     mTSC.calculateSCCs();
-    mMaxSCC.clear();
+    mScanSCC.clear();
     for (auto const& scc : mTSC.mSCCs)
```

Comment:
I would just remove that loop together with mScanSCC class variable and merge it with that main loop in networkEnjoysQuorumIntersection(). Name suggests it is only responsible for building SCCs and the whole thing will be also more easier to understand (as we talked before, you don't need to search for that greatest element). It is not a big deal but current version also iterates all SCCs twice.

Reply:
@fixxxedpoint Well, mScanSCC is used in a couple other helper methods throughout the implementation, I can't easily eliminate it entirely without turning it into a parameter I have to pass everywhere; but I can merge the setup loops as you suggest. Will do so. Otherwise I just wanted to double check: style issues aside, this at least looks _correct_ to you?

Thanks again for all the input and hand-holding; I just want to make sure I'm faithfully transplanting your work here :)

Reply:
(Update: actually making the variable local / sinking it into a member into the helper class was relatively low-impact; patch updated with the results!)